### DISCUSSION TITLE
Action color methods

### DISCUSSION THREAD

**Discussion by Z3d0X** [filamentphp/filament#5119] [Ideas]
```php
// Current:
Action::make()->color('success')

// Proposed new syntax:
Action::make()->success()
```

Notifications also have this syntax
```php
Notification::make()->success()
```

I'd be happy to PR this.
Ref https://github.com/filamentphp/filament/blob/2.x/packages/support/src/Actions/Concerns/HasColor.php

**Comment by danharrin** [comment]
Sure, go for it

**Reply by Wulfheart** [reply]
Has this been PR'ed?

**Reply by Z3d0X** [reply]
It was. However it has been reverted in favour of a better approach in V3. (Usage of Enums) https://github.com/filamentphp/filament/pull/5138

**Reply by Wulfheart** [reply]
Ah. Thanks!
So do I have a way to conditionally give it a color?

**Reply by danharrin** [reply]
`color(fn () => return color here)`, same everywhere
